baybe/searchspace/validation.py

```diff
@@ -39,3 +41,29 @@ def validate_parameters(parameters: Collection[Parameter]) -> None:  # noqa: DOC
 
     # Assert: unique names
     validate_parameter_names(parameters)
+
+
+def get_transform_parameters(
+    parameters: Sequence[Parameter],
+    df: pd.DataFrame,
+    allow_missing: bool,
+    allow_extra: bool,
+):
+    """Extract the parameters relevant for transforming a given dataframe."""
+    parameter_names = [p.name for p in parameters]
+
+    if (not allow_missing) and (missing := set(parameter_names) - set(df)):  # type: ignore[arg-type]
+        raise ValueError(
+            f"The search space parameter(s) {missing} cannot be matched against "
+            f"the provided dataframe. If you want to transform a subset of "
+            f"parameter columns, explicitly set `allow_missing=True`."
+        )
+
+    if (not allow_extra) and (extra := set(df) - set(parameter_names)):
+        raise ValueError(
+            f"The provided dataframe column(s) {extra} cannot be matched against"
+            f"the search space parameters. If you want to transform a dataframe "
+            f"with additional columns, explicitly set `allow_extra=True'."
+        )
+
+    return [p for p in parameters if p.name in df] if allow_missing else parameters
```

Comment:
If I allow missing and allow extra, then this does not return the extra parameters but filters them. Is this intended behavior? If so, we need to make that very clear as I could imagine that some users might think that the extra columns might be kept.

Reply:
Yes, good point. Added the info. However, not here (because in this function, I think it really does not matter), but instead [here](https://github.com/emdgroup/baybe/pull/289#discussion_r1660932690)